code/game/objects/objs.dm

```diff
@@ -208,22 +208,17 @@
 
 /obj/set_dir(ndir)
 	. = ..()
-	if(directional_offset)
-		update_directional_offset()
```

Comment:
Isn't the whole point to update when the dir is changed?

Reply:
The thing is, some items act differently when they're not anchored. 
So they only get offset when they actually are anchored in place, since that's the common case when all things that uses this system actually always want to be offset.